File: WebContent/ddd/simple/view/courseDesign/js/courseDesignProgress/CourseDesignProgressService.js

define(["app"], function (app) {
	app.factory("CourseDesignProgressService", function($resource) {
		
		var courseDesignProgressService = $resource("../DD/CourseDesignProgress/:action", {});
		courseDesignProgressService.saveCourseDesignProgress=function(courseDesignProgress,sucesscb,errorcb)	{
			courseDesignProgressService.save({action:"saveCourseDesignProgress"},{courseDesignProgress:courseDesignProgress},sucesscb,errorcb);
		};
		
		courseDesignProgressService.deleteCourseDesignProgress=function(courseDesignProgressId,sucesscb,errorcb)	{
			courseDesignProgressService.save({action:"deleteCourseDesignProgress"},{courseDesignProgressId:courseDesignProgressId},sucesscb,errorcb);
		};
		
		courseDesignProgressService.updateCourseDesignProgress=function(courseDesignProgress,sucesscb,errorcb) {
			courseDesignProgressService.save({action:"updateCourseDesignProgress"},{courseDesignProgress:courseDesignProgress},sucesscb,errorcb);
		};
		 
		courseDesignProgressService.findCourseDesignProgressById=function(courseDesignProgressId,sucesscb,errorcb) {
			courseDesignProgressService.get({action:"findCourseDesignProgressById",courseDesignProgressId:courseDesignProgressId},sucesscb,errorcb);
		};
		
		courseDesignProgressService.findStage=function(progressStage,courseId,teaId,sucesscb,errorcb){
			courseDesignProgressService.get({action:"findStage",progressStage:progressStage,courseId:courseId,teaId:teaId},sucesscb,errorcb);
		};
		
		courseDesignProgressService.findCourseProgressByCourseIdAndTeaEId=function(courseId,teaId,sucesscb,errorcb) {
			courseDesignProgressService.query({action:"findCourseProgressByCourseIdAndTeaEId",courseId:courseId,teaId:teaId},sucesscb,errorcb);
		};
		
		courseDesignProgressService.findAllCourseDesignProgress=function(sucesscb,errorcb) {
			courseDesignProgressService.query({action:"findAllCourseDesignProgress"},sucesscb,errorcb);
		};
		return courseDesignProgressService;
})});

File: src/ddd/simple/service/courseDesign/CourseDesignProgressService.java
package ddd.simple.service.courseDesign;

import ddd.base.persistence.EntitySet;
import ddd.simple.service.base.BaseServiceInterface;
import ddd.simple.entity.courseDesign.CourseDesignProgress;

public interface CourseDesignProgressService extends BaseServiceInterface {
	public CourseDesignProgress saveCourseDesignProgress(CourseDesignProgress courseDesignProgress) ;
	
	public int deleteCourseDesignProgress(Long courseDesignProgressId) ;
	
	public CourseDesignProgress updateCourseDesignProgress(CourseDesignProgress courseDesignProgress) ;
	
	public CourseDesignProgress findCourseDesignProgressById(Long courseDesignProgressId) ;
	
	public EntitySet<CourseDesignProgress> findCourseProgressByCourseIdAndTeaEId(Long courseId,Long teaId);
	
	public CourseDesignProgress findStage(int progressStage,Long courseId,Long teaId);
	
	public EntitySet<CourseDesignProgress> findAllCourseDesignProgress() ;
 
}